src/components/Application/ApplicationContainer.tsx

```diff
@@ -10,12 +10,17 @@ export interface ApplicationContainerProps {
 
 const ApplicationContainer: React.FC<ApplicationContainerProps> = ({ template, children }) => {
   const { t } = useLanguageProvider()
-  const { replace } = useRouter()
+  const { push, location } = useRouter()
   const {
     userState: { currentUser, isNonRegistered },
   } = useUserState()
+  const [prevQueryString] = useState(location?.state?.prevQuery)
```

Comment:
This shouldn't really need to be a `useState` variable, but the re-renders mess it up. You'll see that the same thing is already done for the Dataview table (going back from the DataView detail page), and we didn't need any extra `useState` there.

One day we can go and hunt down some of these unnecessary re-renders....